src/Symfony/Bundle/SecurityBundle/ArgumentResolver/UserArgumentResolver.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+namespace Symfony\Bundle\SecurityBundle\ArgumentResolver;
+
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Controller\ArgumentResolver\ArgumentResolverInterface;
+use Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorageInterface;
+
+/**
+ * @author Wouter J <wouter@wouterj.nl>
+ */
+class UserArgumentResolver implements ArgumentResolverInterface
```

Comment:
Hmm, do you mean using the Doctrine Param Converter? That's why I made this resolver only work with `UserInterface`. When using the Doctrine Param Converter, one would actually need to typehint to a specific implementation.

Btw, this may be a good reason to implement priority for argument resolvers.